xbmc/utils/ContentUtils.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ContentUtils.h"
+
+#include "FileItem.h"
+#include "video/VideoInfoTag.h"
+
+const std::string ContentUtils::GetPreferredArtImage(const CFileItem& item)
+{
+  // movies
+  if (item.HasVideoInfoTag() && item.GetVideoInfoTag()->m_type == MediaTypeMovie &&
+      item.HasArt("poster"))
+  {
+    return item.GetArt("poster");
+  }
+
+  // shows
+  else if (item.HasVideoInfoTag() && item.GetVideoInfoTag()->m_type == MediaTypeTvShow &&
+           item.HasArt("poster"))
+  {
+    return item.GetArt("poster");
+  }
+
+  // seasons
+  else if (item.HasVideoInfoTag() && item.GetVideoInfoTag()->m_type == MediaTypeSeason &&
+           item.HasArt("poster"))
+  {
+    return item.GetArt("poster");
+  }
```

Comment:
This feels very complex. If there was a `HasPreferredArtMediaType()` function only a single if condition would be required.

Reply:
You mean an helper function that would return the preferred art type provided the mediatype?

Reply:
Like:

```cpp
const std::string ContentUtils::GetPreferredArtImage(const CFileItem& item)
{
  // movies, tvshow or season
  if (item.HasVideoInfoTag() && HasPreferredArtMediaType(item) &&
      item.HasArt("poster"))
  {
    return item.GetArt("poster");
  }

  return item.GetArt("thumb");
}

bool ContentUtils:: HasPreferredArtMediaType(const CFileItem& item)
{
  return item.GetVideoInfoTag()->m_type == MediaTypeMovie ||
             item.GetVideoInfoTag()->m_type == MediaTypeTvShow ||
             item.GetVideoInfoTag()->m_type == MediaTypeSeason;
}

Reply:
I see your point, however this approach is not extensible as you're setting always the "poster" type to any mediatype that needs special handling. I'll try to work something out on the same line of your approach

Reply:
👍